reframe/frontend/cli.py

```diff
@@ -581,10 +653,26 @@ def main():
     printer.debug(format_env(options.env_vars))
 
     # Setup the check loader
+    if options.retry_failed is not None:
+        if options.retry_failed:
+            filename = options.retry_failed
+        else:
+            filename = os_ext.expandvars(
+                site_config.get('general/report_file'))
```

Comment:
This is wrong, because you don't take into account the case that the `report_file` has the `{sessionid}` pattern in it. You should use the `generate_report_filename()` function somehow.